### DISCUSSION TITLE
Welcome to sfnetworks Discussions!

### DISCUSSION THREAD

**Discussion by luukvdmeer** [luukvdmeer/sfnetworks#97] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Start a debate about pros and cons of current functionalities in the package.
  * Share ideas.
  * Share experiences.
  * Ask questions you’re wondering about.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

Please don't hesitate to open a discussion topic!

**Comment by Robinlovelace** [comment]
Hi everyone, great to see a place for wider discussions rather than the more focussed issues to fix that are well documented in the issue tracker. Just wanted to say and that if you're interested in using spatial networks for transport planning and have questions I look forward to them! I develop the `stplanr` package and hope to eventually use `sfnetworks` for local routing based on OSM data.

**Comment by agila5** [comment]
Hi everyone! Looking forward to great discussions and ideas 🎉

**Comment by marcoscarloseduardo** [comment]
Hi everyone, I would like to know if possible that in vignette 2, the paragraph: "...the contraction of a set of nodes P={p1,p2,...,pn} is the replacement of S and all its incident edges..." in fact  it is "...the contraction of a set of nodes P={p1,p2,...,pn} is the replacement of **P** and all its incident edges...".

I'm working in garbage truck routing optimization in a city in Argentina. I'm learning the sfnetworks package to fix some aspects of the topology I download from OSM. I think that the spatial morphs to unify nodes and reduce edges will be useful for me.

Based on your experience, what R packages could help me solve the routing taking into account the capacity of the trucks and minimizing the distances and number of turns they make in the path?

Thank in advance,
Carlos.

**Comment by agila5** [comment]
Hi @marcoscarloseduardo! First of all, thank you very much for your comment. 

> I would like to know if possible that in vignette 2, the paragraph: "...the contraction of a set of nodes P={p1,p2,...,pn} is the replacement of S and all its incident edges..." in fact it is "...the contraction of a set of nodes P={p1,p2,...,pn} is the replacement of P and all its incident edges...".

The vignette was written by @luukvdmeer but I think you are right. If you want, feel free to create a PR to the _develop_ branch and fix that typo. I can add here simple instructions if you want. 

> Based on your experience, what R packages could help me solve the routing taking into account the capacity of the trucks and minimizing the distances and number of turns they make in the path?

I don't have a lot of experience with these types of problems, but I will try to play with some examples at the end of this week (and maybe add here something if I find any useful approach). In any case, I think you could create a new discussion and summarise there all the relevant details. 

Again, thanks for your comments.

**Reply by marcoscarloseduardo** [reply]
Hi @agila5, ok, yes let me know the instructions to learn how to create a PR.

And I will appreciate any help and recommendations you can give me on this, especially if there is something to take into account the turns with the graph's own georeferenced information. Otherwise I imagine I will have to split each node into a trident with additional weights on those that turn left and right but that is going to add complexity to the processing.

Thank you very much,
Carlos.

**Comment by agila5** [comment]
> Hi @agila5, ok, yes let me know the instructions to learn how to create a PR.

See [here](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) for generic instructions and check [this](https://happygitwithr.com/) book for more details on how to integrate git and R/Rstudio.
